pkg/filter/run.go

```diff
@@ -0,0 +1,84 @@
+package filter
+
+import (
+	"github.com/shipwright-io/build/pkg/apis/build/v1alpha1"
+	"github.com/shipwright-io/triggers/pkg/constants"
+
+	tknv1alpha1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1"
+	tknv1beta1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// TektonRunParamsToShipwrightParamValues transforms the informed Tekton Run params into Shipwright
+// ParamValues slice.
+func TektonRunParamsToShipwrightParamValues(run *tknv1alpha1.Run) []v1alpha1.ParamValue {
+	paramValues := []v1alpha1.ParamValue{}
+	for _, p := range run.Spec.Params {
+		paramValue := v1alpha1.ParamValue{Name: p.Name}
+		if p.Value.Type == tknv1beta1.ParamTypeArray {
+			paramValue.Values = []v1alpha1.SingleValue{}
+			for _, v := range p.Value.ArrayVal {
+				v := v
+				paramValue.Values = append(paramValue.Values, v1alpha1.SingleValue{
+					Value: &v,
+				})
+			}
+		} else {
```

Comment:
Tekton has recently added ParamTypeObject. I guess we want to reject this.

Reply:
Do you have more examples about it? Thanks in advance.

Reply:
The Tekton TEP is https://github.com/tektoncd/community/blob/main/teps/0075-object-param-and-result-types.md.

The general purpose is to better group many related parameters into one.

We don't support them in Shipwright and I at the moment do not yet see the need for them anyway because a Build is just one step of a larger pipeline and does rarely have groups of parameters. Vs. in a larger Tekton pipeline you could have parameters related to source code retrieval (Git repo, Git branch), build related (context directory, Dockerfile if we stick with image builds as an example), and maybe also deployment related (kube namespace, etc.)

Your code should have an `else if` checking for the explicit type `ParamTypeString` and an else block with things we do not support. The whole `if` might then be rewritten as a `switch` but that's up to what you prefer.